used the following config:

```json
{
   "dataPipelineId": "crossref_event_data_web_api_10",
   "gcpProjectName": "elife-data-pipeline",
   "importedTimestampFieldName": "imported_timestamp",
   "dataset": "de_dev",
   "table": "crossref_event_data_web_api",
   "stateFile": {
      "bucketName": "ci-elife-data-pipeline",
      "objectName": "by-env/de_dev/airflow-config/generic-web-api/de_dev-crossref-event-data-state-10.1101.json"
   },
   "dataUrl": {
      "urlExcludingConfigurableParameters": "https://api.eventdata.crossref.org/v1/events?rows=100&mailto=h.ciplak@elifesciences.org&obj-id.prefix=10.1101",
      "configurableParameters": {
         "nextPageCursorParameterName": "cursor",
         "fromDateParameterName": "from-collected-date",
         "toDateParameterName": "until-collected-date",
         "daysDiffFromStartTillEnd": 1,
         "defaultStartDate": "2018-01-01",
         "dateFormat": "%Y-%m-%d"
      }
   },
   "response": {
      "nextPageCursorKeyFromResponseRoot": [
         "message",
         "next-cursor"
      ],
      "itemsKeyFromResponseRoot": [
         "message",
         "events"
      ],
      "totalItemsCountKeyFromResponseRoot": [
         "message",
         "total-results"
      ],
      "recordTimestamp": {
         "itemTimestampKeyFromItemRoot": [
            "timestamp"
         ]
      }
   }
}
```

Now getting the following error for, unrelated to the change:

```text
[2022-12-28, 10:30:16 UTC] {generic_web_api_data_etl.py:99} INFO - Request URL: https://api.eventdata.crossref.org/v1/events?rows=100&mailto=h.ciplak@elifesciences.org&obj-id.prefix=10.1101&from-collected-date=2018-01-12&cursor=6ffa61ec-b8fe-48b0-81b4-ae551f94a5de&until-collected-date=2018-01-13
[2022-12-28, 10:30:16 UTC] {generic_web_api_data_etl.py:108} INFO - Headers: MappingConfig(printable_mapping={})
[2022-12-28, 10:30:17 UTC] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:1045: InsecureRequestWarning: Unverified HTTPS request is being made to host 'api.eventdata.crossref.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(

[2022-12-28, 10:30:18 UTC] {generate_schema.py:195} INFO - Processing line 1000
[2022-12-28, 10:30:19 UTC] {generate_schema.py:195} INFO - Processing line 2000
[2022-12-28, 10:30:19 UTC] {generate_schema.py:218} INFO - Processed 2946 lines
[2022-12-28, 10:30:33 UTC] {taskinstance.py:1851} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 193, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/web_api_data_import_pipeline.py", line 40, in web_api_data_etl
    generic_web_api_data_etl(
  File "/opt/airflow/data_pipeline/generic_web_api/generic_web_api_data_etl.py", line 241, in generic_web_api_data_etl
    process_web_api_data_etl_batch(
  File "/opt/airflow/data_pipeline/generic_web_api/generic_web_api_data_etl.py", line 208, in process_web_api_data_etl_batch
    load_written_data_to_bq(data_config, full_temp_file_location)
  File "/opt/airflow/data_pipeline/generic_web_api/generic_web_api_data_etl.py", line 268, in load_written_data_to_bq
    load_file_into_bq(
  File "/opt/airflow/data_pipeline/utils/data_store/bq_data_service.py", line 56, in load_file_into_bq
    job.result()
  File "/home/airflow/.local/lib/python3.8/site-packages/google/cloud/bigquery/job/base.py", line 708, in result
    return super(_AsyncJob, self).result(timeout=timeout, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/google/api_core/future/polling.py", line 137, in result
    raise self._exception
google.api_core.exceptions.BadRequest: 400 Error while reading data, error message: JSON table encountered too many errors, giving up. Rows: 2674; errors: 1. Please look into the errors[] collection for more details.
```

--- a/data_pipeline/generic_web_api/generic_web_api_data_etl.py
+++ b/data_pipeline/generic_web_api/generic_web_api_data_etl.py
@@ -228,11 +228,11 @@ def generic_web_api_data_etl(
     data_config: WebApiConfig,
     end_timestamp: Optional[datetime] = None
 ):
-    if not end_timestamp:
-        end_timestamp = get_current_timestamp()
-    LOGGER.info('end_timestamp (if used): %r', end_timestamp)
     stored_state = get_stored_state(data_config)
     current_from_timestamp = stored_state
+    if not end_timestamp and current_from_timestamp:
+        end_timestamp = get_current_timestamp()
+    LOGGER.info('end_timestamp: %r', end_timestamp)
     while True:
         next_from_timestamp = get_next_batch_from_timestamp_for_config(
             data_config=data_config,

--- a/data_pipeline/generic_web_api/generic_web_api_data_etl.py
+++ b/data_pipeline/generic_web_api/generic_web_api_data_etl.py
@@ -42,7 +42,7 @@ from data_pipeline.utils.data_pipeline_timestamp import (
 LOGGER = logging.getLogger(__name__)
 
 
-def get_stored_state(
+def get_start_timestamp_from_state_file_or_optional_default_value(
         data_config: WebApiConfig,
 ):
     try:
@@ -228,7 +228,7 @@ def generic_web_api_data_etl(
     data_config: WebApiConfig,
     end_timestamp: Optional[datetime] = None
 ):
-    stored_state = get_stored_state(data_config)
+    stored_state = get_start_timestamp_from_state_file_or_optional_default_value(data_config)
     current_from_timestamp = stored_state
     if not end_timestamp and current_from_timestamp:
         end_timestamp = get_current_timestamp()

--- a/tests/unit_test/generic_web_api/generic_web_api_data_etl_test.py
+++ b/tests/unit_test/generic_web_api/generic_web_api_data_etl_test.py
@@ -461,3 +461,20 @@ class TestGenericWebApiDataEtl:
             for call_args in get_data_single_page_mock.call_args_list
         ]
         assert actual_from_and_until_date_list == expected_from_and_until_date_list
+
+    def test_should_pass_none_from_and_until_dates_if_not_configured(
+        self,
+        get_stored_state_mock: MagicMock,
+        get_data_single_page_mock: MagicMock
+    ):
+        expected_from_and_until_date_list = [(None, None)]
+        data_config = get_data_config(WEB_API_CONFIG)
+        get_stored_state_mock.return_value = None
+        item_list = [{'key': 'value'}]
+        get_data_single_page_mock.return_value = item_list
+        generic_web_api_data_etl(data_config)
+        actual_from_and_until_date_list = [
+            (call_args.kwargs['from_date'], call_args.kwargs['until_date'])
+            for call_args in get_data_single_page_mock.call_args_list
+        ]
+        assert actual_from_and_until_date_list == expected_from_and_until_date_list

--- a/tests/unit_test/generic_web_api/generic_web_api_data_etl_test.py
+++ b/tests/unit_test/generic_web_api/generic_web_api_data_etl_test.py
@@ -33,10 +33,14 @@ def _process_downloaded_data_mock():
         yield mock
 
 
-@pytest.fixture(name='get_stored_state_mock', autouse=True)
-def _get_stored_state_mock():
+@pytest.fixture(
+    name='get_start_timestamp_from_state_file_or_optional_default_value_mock',
+    autouse=True
+)
+def _get_start_timestamp_from_state_file_or_optional_default_value_mock():
     with patch.object(
-            generic_web_api_data_etl_module, 'get_stored_state'
+        generic_web_api_data_etl_module,
+        'get_start_timestamp_from_state_file_or_optional_default_value'
     ) as mock:
         yield mock
 
@@ -426,7 +430,7 @@ class TestGenericWebApiDataEtl:
 
     def test_should_retrieve_data_in_date_range_batches(
         self,
-        get_stored_state_mock: MagicMock,
+        get_start_timestamp_from_state_file_or_optional_default_value_mock: MagicMock,
         get_data_single_page_mock: MagicMock
     ):
         timestamp_string_1 = '2020-01-01+00:00'
@@ -452,7 +456,9 @@ class TestGenericWebApiDataEtl:
                 default_start_date=timestamp_string_1
             )
         )
-        get_stored_state_mock.return_value = initial_timestamp
+        get_start_timestamp_from_state_file_or_optional_default_value_mock.return_value = (
+            initial_timestamp
+        )
         item_list = [{'timestamp': timestamp_string_2}]
         get_data_single_page_mock.return_value = item_list
         generic_web_api_data_etl(data_config, end_timestamp=end_timestamp)
@@ -464,12 +470,12 @@ class TestGenericWebApiDataEtl:
 
     def test_should_pass_none_from_and_until_dates_if_not_configured(
         self,
-        get_stored_state_mock: MagicMock,
+        get_start_timestamp_from_state_file_or_optional_default_value_mock: MagicMock,
         get_data_single_page_mock: MagicMock
     ):
         expected_from_and_until_date_list = [(None, None)]
         data_config = get_data_config(WEB_API_CONFIG)
-        get_stored_state_mock.return_value = None
+        get_start_timestamp_from_state_file_or_optional_default_value_mock.return_value = None
         item_list = [{'key': 'value'}]
         get_data_single_page_mock.return_value = item_list
         generic_web_api_data_etl(data_config)

